lua_modules/http/httpserver.lua

```diff
@@ -79,17 +79,19 @@ do
   ------------------------------------------------------------------------------
   local http_handler = function(handler)
     return function(conn)
+      local req, res
+      local buf = ""
+      local method, url
       local csend = (require "fifosock").wrap(conn)
       local cfini = function()
         conn:on("receive", nil)
         conn:on("disconnection", nil)
-        csend(function() conn:on("sent", nil) conn:close() end)
+        csend(function() conn:on("sent", nil) conn:close() res = nil req = nil end)
       end
-      local req, res
-      local buf = ""
-      local method, url
       local ondisconnect = function(conn)
-	conn.on("sent", nil)
+        conn.on("sent", nil)
+        res = nil
+        req = nil
```

Comment:
1. Does `ondisconnect` really need to take the `conn` as an argument?
2. I think `conn.on()` is an error and it should be `conn:on()`?
3. Assuming that the answer to 1 is no and that 2 is true, then move `ondisconnect` above and have `cfini` call `csend(ondisconnect)` rather than duplicating code.

Reply:
I am not sure. in fact, when the disconnect event came, call this `ondisconnect` will reboot. 

I think that  should set the `res` and `req` to nil to avoid memory leak in this disconnect callback.

This `ondisconnect` had before. so, as few changes as possible

Reply:
I think the reason for the reboot is that `conn.on` should be `conn:on`.  Please test with that change in place, and if it solves the problem, please add it to the commit stack for this PR.

Reply:
OK, let me try

Reply:
I think that points 1 and 3 still stand here?

Reply:
The `ondisconnect(conn)` is need. lost the `conn`, the menory will leak.

Reply:
OK.